### PR TITLE
Rsz update longe wire buffer insertion

### PR DESCRIPTION
**Pull request by arthurjolo**
Fixes [#4006](https://github.com/The-OpenROAD-Project/OpenROAD/issues/4006)

This PR changes the position of buffers inserted to fix long wires.

The new position is set to be the the middle of the wire, previously it was set to be on the max wire length allowed causing some buffers to be inserted near the drivers.

### CODE REVIEW

**Review by maliberty — APPROVED** [review]

File: src/rsz/src/RepairDesign.cc [hunk 1]
```diff
@@ -851,7 +851,8 @@ void RepairDesign::repairNetWire(
                  level,
                  units_->distanceUnit()->asString(dbuToMeters(wire_length), 1),
                  units_->distanceUnit()->asString(dbuToMeters(max_length_), 1));
-      split_length = min(split_length, max(max_length_ - wire_length_ref, 0));
+      split_length = min(max_length_, length / 2);
```
Comment (povik) [inline src/rsz/src/RepairDesign.cc:854]: This reintroduces what was fixed in 3b97c566

Comment (maliberty) [reply src/rsz/src/RepairDesign.cc:854]: @arthurjolo please comment

Comment (arthurjolo) [reply src/rsz/src/RepairDesign.cc:854]: I believe this was overwritten during a merge. But I think it easy to fix: `split_length = min(length / 2, max(max_length_ - wire_length_ref, 0));`
This should account for both changes, @povik do you agree?

Comment (povik) [reply src/rsz/src/RepairDesign.cc:854]: Yes, that looks good to me! (I was lazy to type that out myself...)

Comment (maliberty) [reply src/rsz/src/RepairDesign.cc:854]: @arthurjolo please make a PR for your fix.

**Pull request merged**